### PR TITLE
Update ChatModel response with defaults

### PR DESCRIPTION
**Pull request by daniellok-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/daniellok-db/mlflow/pull/12298?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12298/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12298
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Addressing recent feedback that ChatModel responses contain required fields that users might not be able to fill out. This PR updates the types with the suggested defaults.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/types/llm.py [hunk 1]
```diff
@@ -223,12 +223,13 @@ class ChatChoice(_BaseDataclass):
...
         message (:py:class:`ChatMessage`): The message that was generated.
         finish_reason (str): The reason why generation stopped.
+            **Optional**, defaults to ``"stop"``
         logprobs (:py:class:`ChatChoiceLogProbs`): Log probability information for the choice.
     """
 
     index: int
     message: ChatMessage
-    finish_reason: str
+    finish_reason: str = "stop"
```
Comment (WeichenXu123) [inline mlflow/types/llm.py:232]: Q:

Is it risky to assume the default finish reason to be "stop" ?
sometimes the finish reason is lost but actual finish reason might be max-token limit or other reasons

**Review by WeichenXu123 — COMMENTED** [review]
LGTM except one issue.

File: mlflow/types/llm.py (hunk 1 above)
Comment (daniellok-db) [reply mlflow/types/llm.py:232]: Yeah I think this is possible, though I think the intention of this PR is just to provide reasonable defaults when users don't have the info available. If they're integrating directly with openai or transformers, they can just put the actual finish reason in here.

File: mlflow/types/llm.py [hunk 2]
```diff
@@ -255,18 +256,21 @@ class TokenUsageStats(_BaseDataclass):
...
         total_tokens (int): The total number of tokens used.
+            **Optional**, defaults to ``None``
     """
 
-    prompt_tokens: int
-    completion_tokens: int
-    total_tokens: int
+    prompt_tokens: int = None
+    completion_tokens: int = None
+    total_tokens: int = None
```
Comment (harupy) [inline mlflow/types/llm.py:268]:
```suggestion
    prompt_tokens: Optional[int] = None
    completion_tokens: Optional[int] = None
    total_tokens: Optional[int] = None
```

File: mlflow/types/llm.py [hunk 3]
```diff
@@ -275,28 +279,28 @@ class ChatResponse(_BaseDataclass):
...
             containing the generated responses
         usage (:py:class:`TokenUsageStats`): An object describing the tokens used by the request.
     """
 
-    id: str
-    model: str
     choices: List[ChatChoice]
     usage: TokenUsageStats
+    id: str = None
+    model: str = None
```
Comment (harupy) [inline mlflow/types/llm.py:295]:
```suggestion
    id: Optional[str] = None
    model: Optional[str] = None
```

File: mlflow/types/llm.py [hunk 4]
```diff
@@ -255,18 +256,21 @@ class TokenUsageStats(_BaseDataclass):
 
     Args:
         prompt_tokens (int): The number of tokens in the prompt.
+            **Optional**, defaults to ``None``
```
Comment (harupy) [inline mlflow/types/llm.py:259]: I'd remove this because the type hint indicates that but fine to keep it.

**Review by harupy — APPROVED** [review]
Left a few comments, otherwise LGTM!

**Pull request merged**